### PR TITLE
New noise & socks5 APIs

### PR DESCRIPTION
**Pull request by dr-orlovsky**

### CODE REVIEW

**Comment by dr-orlovsky** [comment]
ACK f24b97498560e76d7d384986a6d5f8ee0f93d544

File: radicle-node/src/wire/transport.rs [hunk 1]
```diff
@@ -131,15 +135,15 @@ pub struct Transport<R, S, W, G: Negotiator> {
     /// Backing service instance.
     service: Service<R, S, W, G>,
     /// Worker pool interface.
-    worker: chan::Sender<WorkerReq<G>>,
+    worker: chan::Sender<WorkerReq>,
     /// Used to performs X25519 key exchange.
-    keypair: G,
+    node_sk: cyphernet::crypto::ed25519::PrivateKey,
```
Comment (cloudhead) [inline radicle-node/src/wire/transport.rs:140]: I don't like this much, why can't we hold a signer like before?

File: radicle-node/src/client.rs [hunk 2]
```diff
@@ -103,10 +104,14 @@ impl<G: crypto::Signer + crypto::Negotiator + Clone + 'static> Runtime<G> {
             rng,
         );
 
+        let node_id = signer.public_key().0;
+        let sig = signer.sign(node_id.as_slice());
+        let auth = Authenticator::new(node_id.into(), sig.0.into());
+
```
Comment (cloudhead) [inline radicle-node/src/client.rs:110]: We'll need a nonce too actually!

**Comment by cloudhead** [comment]
Amazing, it works!

File: radicle-node/src/client.rs (hunk 2 above)
Comment (cloudhead) [reply radicle-node/src/client.rs:110]: To save a round-trip, maybe the nonce could be the noise session hash :thinking:

**Pull request closed without merging**